backend/src/main/java/de/neuefische/booklibrary/backend/security/AppUserController.java

```diff
@@ -0,0 +1,88 @@
+package de.neuefische.booklibrary.backend.security;
+
+import de.neuefische.booklibrary.backend.SecurityConfig;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import javax.servlet.http.HttpSession;
+import javax.validation.Valid;
+
+@RestController
+@RequestMapping("/api/app-users")
+@RequiredArgsConstructor
+public class AppUserController {
+
+    private final AppUserService appUserService;
+
+    @GetMapping("/login")
+    public HttpStatus login() {
+        return HttpStatus.OK;
+
+    }
+
+    @GetMapping("/me")
+    public String me() {
+        return SecurityContextHolder
+                .getContext()
+                .getAuthentication()
+                .getName();
+    }
+
+    @GetMapping("/user")
+    public AppUserInfo getAppUser() {
+        String name = SecurityContextHolder
+                .getContext()
+                .getAuthentication()
+                .getName();
+        return appUserService.getUserInfo(name);
+    }
+
+    @GetMapping("/role")
+    public String getRole() {
+        return SecurityContextHolder
+                .getContext()
+                .getAuthentication()
+                .getAuthorities()
+                .toString();
+    }
```

Comment:
Now you have 3 Endpoints: 1 with name, 1 with role, and 1 with name&role 😄 still not sure if 3 are really necessary, but if you're fine with it its okay. I don't want you to change everything again

Reply:
Ja du hast Recht. Ich werde dies vielleicht später refaktorisieren.